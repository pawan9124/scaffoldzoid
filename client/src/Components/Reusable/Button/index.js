import React from "react";
import PropTypes from "prop-types";
import "./style.css";

function Button(props) {
  return (
    <button
      data-test="buttonComponent"
      className={`btn  ${props.buttonClass}`}
      onClick={props.onClick}
      data-toggle={props.dataToggle}
      data-target={props.dataTarget}
      data-dismiss={props.dataDismiss}
      id={props.id}
    >
      {props.buttonLabel}
      <span className="btn-icon">{props.icon}</span>
    </button>
  );
}

Button.propTypes = {
  buttonClass: PropTypes.string.isRequired,
  buttonLabel: PropTypes.string.isRequired,
};

export default Button;
