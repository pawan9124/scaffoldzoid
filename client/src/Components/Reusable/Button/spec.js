import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../../Utils";
import Button from "./index";
import renderer from "react-test-renderer";

describe("Button Component", () => {
  describe("Taking snap shot", () => {
    it("Taking the snapshot of the Button Component", () => {
      let mockFunc = jest.fn();
      const props = {
        buttonClass: "btn-primary",
        buttonLabel: "Test Action",
        onClick: mockFunc,
      };
      const tree = renderer.create(<Button {...props} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("Checking PropTypes", () => {
    it("Should Not throw a warning", () => {
      const expectedProps = {
        buttonClass: "btn-primary",
        buttonLabel: "Test Action",
        onClick: () => {},
      };

      const propsError = checkProps(Button, expectedProps);
      expect(propsError).toBeUndefined();
    });
  });

  describe("Renders", () => {
    let wrapper;
    let mockFunc;
    beforeEach(() => {
      mockFunc = jest.fn();
      const props = {
        buttonClass: "btn-primary",
        buttonLabel: "Test Action",
        onClick: mockFunc,
      };

      wrapper = shallow(<Button {...props} />);
    });

    it("Should Render a button", () => {
      const button = findByTestAttr(wrapper, "buttonComponent");
      expect(button.length).toBe(1);
    });

    it("Should Emit callback on click event", () => {
      const button = findByTestAttr(wrapper, "buttonComponent");
      button.simulate("click");
      const callback = mockFunc.mock.calls.length;
      expect(callback).toBe(1);
    });
  });
});
