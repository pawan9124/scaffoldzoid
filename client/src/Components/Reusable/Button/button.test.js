import React from "react";
import { shallow } from "enzyme";
import { findByTestAttr, checkProps } from "../../../testUtils";
import Button from "./index";
import renderer from "react-test-renderer";

let mockFunc = jest.fn();
const props = {
  buttonClass: "btn-primary",
  buttonLabel: "Test Action",
  onClick: mockFunc,
};

describe("Taking snap shot", () => {
  it("Taking the snapshot of the Button Component", () => {
    const tree = renderer.create(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Checking PropTypes", () => {
  it("Should Not throw a warning", () => {
    const expectedProps = {
      buttonClass: "btn-primary",
      buttonLabel: "Test Action",
      onClick: () => {},
    };

    const propsError = checkProps(Button, expectedProps);
    expect(propsError).toBeUndefined();
  });
});

describe("Checking the render of button", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Button {...props} />);
  });
  it("Should Render a button", () => {
    const button = findByTestAttr(wrapper, "buttonComponent");
    expect(button.length).toBe(1);
  });
});
