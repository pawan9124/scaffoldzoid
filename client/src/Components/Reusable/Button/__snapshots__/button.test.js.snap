// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taking snap shot Taking the snapshot of the Button Component 1`] = `
<button
  className="btn  btn-primary"
  data-test="buttonComponent"
  onClick={[MockFunction]}
>
  Test Action
  <span
    className="btn-icon"
  />
</button>
`;
